addons/l10n_ar_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,199 @@
+        #   - Net amount total
+        wizard = self.env['account.payment.register'].with_context(active_model='account.move.line', active_ids=self.line_ids.ids).new()
+        wizard.currency_id = self.l10n_latam_check_id.currency_id
+        wizard.payment_date = self.payment_date
+        wizard.withholding_ids = [Command.clear()] + [Command.create({'tax_id': x.tax_id.id, 'base_amount': x.base_amount , 'amount': x.amount}) for x in self.withholding_ids]
+        return (wizard.amount / wizard.net_amount, wizard.net_amount)
+
+    def _get_withholding_move_line_default_values(self):
+        return {
+            'partner_id': self.partner_id.id,
+            'currency_id': self.currency_id.id,
+        }
+
+    def _create_payment_vals_from_wizard(self, batch_result):
+        payment_vals = super()._create_payment_vals_from_wizard(batch_result)
+        payment_vals['amount'] = self.net_amount
+        conversion_rate = self._get_conversion_rate()
+        sign = 1
+        if self.partner_type == 'supplier':
+            sign = -1
+        for line in self.withholding_ids:
+            if not line.name or line.name == '/':
+                if line.tax_id.l10n_ar_withholding_sequence_id:
+                    line.name = line.tax_id.l10n_ar_withholding_sequence_id.next_by_id()
+                else:
+                    raise UserError(_('Please enter withholding number for tax %s') % line.tax_id.name)
+            dummy, account_id, tax_repartition_line_id = line._tax_compute_all_helper()
+            balance = self.company_currency_id.round(line.amount * conversion_rate)
+            payment_vals['write_off_line_vals'].append({
+                    **self._get_withholding_move_line_default_values(),
+                    'name': line.name,
+                    'account_id': account_id,
+                    'amount_currency': sign * line.amount,
+                    'balance': sign * balance,
+                    'tax_base_amount': sign * line.base_amount,
+                    'tax_repartition_line_id': tax_repartition_line_id,
+            })
+
+        for base_amount in list(set(self.withholding_ids.mapped('base_amount'))):
+            withholding_lines = self.withholding_ids.filtered(lambda x: x.base_amount == base_amount)
+            nice_base_label = ','.join(withholding_lines.mapped('name'))
+            account_id = self.company_id.l10n_ar_tax_base_account_id.id
+            base_amount = sign * base_amount
+            cc_base_amount = self.company_currency_id.round(base_amount * conversion_rate)
+            payment_vals['write_off_line_vals'].append({
+                **self._get_withholding_move_line_default_values(),
+                'name': _('Base Ret: ') + nice_base_label,
+                'tax_ids': [Command.set(withholding_lines.mapped('tax_id').ids)],
+                'account_id': account_id,
+                'balance': cc_base_amount,
+                'amount_currency': base_amount,
```

Comment:
```suggestion
                'amount_currency': base_amount,
```  
Let the system do its job.

Reply:
If I remove balance I get this error


2023-10-03 22:33:23,641 977 ERROR l10n_ar_withholding odoo.http: Exception during request handling. 
Traceback (most recent call last):
  File "/home/odoo/custom/odoo/odoo/http.py", line 2149, in __call__
    response = request._serve_db()
  File "/home/odoo/custom/odoo/odoo/http.py", line 1724, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/home/odoo/custom/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/home/odoo/custom/odoo/odoo/http.py", line 1751, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/home/odoo/custom/odoo/odoo/http.py", line 1952, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/home/odoo/custom/odoo/odoo/addons/base/models/ir_http.py", line 191, in _dispatch
    result = endpoint(**request.params)
  File "/home/odoo/custom/odoo/odoo/http.py", line 719, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/home/odoo/custom/odoo/addons/web/controllers/dataset.py", line 28, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/home/odoo/custom/odoo/addons/web/controllers/dataset.py", line 20, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo/custom/odoo/odoo/api.py", line 461, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/home/odoo/custom/odoo/odoo/api.py", line 448, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/custom/odoo/addons/account/wizard/account_payment_register.py", line 938, in action_create_payments
    payments = self._create_payments()
  File "/home/odoo/custom/odoo/addons/account/wizard/account_payment_register.py", line 932, in _create_payments
    payments = self._init_payments(to_process, edit_mode=edit_mode)
  File "/home/odoo/custom/odoo/addons/account/wizard/account_payment_register.py", line 797, in _init_payments
    .create([x['create_vals'] for x in to_process])
  File "<decorator-gen-191>", line 2, in create
  File "/home/odoo/custom/odoo/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/odoo/custom/odoo/addons/account/models/account_payment.py", line 726, in create
    to_write['line_ids'] = [(0, 0, line_vals) for line_vals in pay._prepare_move_line_default_vals(write_off_line_vals=write_off_line_vals)]
  File "/home/odoo/custom/odoo/addons/l10n_latam_check/models/account_payment.py", line 272, in _prepare_move_line_default_vals
    res = super()._prepare_move_line_default_vals(write_off_line_vals=write_off_line_vals)
  File "/home/odoo/custom/odoo/addons/account/models/account_payment.py", line 297, in _prepare_move_line_default_vals
    write_off_balance = sum(x['balance'] for x in write_off_line_vals_list)
  File "/home/odoo/custom/odoo/addons/account/models/account_payment.py", line 297, in <genexpr>
    write_off_balance = sum(x['balance'] for x in write_off_line_vals_list)
KeyError: 'balance'

Reply:
unsertood, it's related to account.move being created from payments